### PR TITLE
Type hinting for Factory.php up to PHPStan 9

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Closes #

Changes proposed in this pull request:

- Type hinting for Factory.php up to PHPStan 9

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested

### CODE REVIEW

File: app/Models/EntryDAO.php [hunk 1]
```diff
@@ -1197,10 +1197,15 @@ public function listByIds(array $ids, string $order = 'DESC') {
 
 	/**
 	 * For API
-	 * @return array<string>
+	 * @return array<string>|false
 	 */
-	public function listIdsWhere($type = 'a', $id = '', $state = FreshRSS_Entry::STATE_ALL,
-			$order = 'DESC', $limit = 1, $firstId = '', $filters = null): array {
+	public function listIdsWhere($type = 'a',
```
Comment (Alkarex) [inline app/Models/EntryDAO.php]: Why no types here?

File: app/Models/Factory.php [hunk 2]
```diff
@@ -46,7 +70,12 @@ public static function createTagDao($username = null) {
 		}
 	}
 
-	public static function createStatsDAO($username = null) {
+	/**
+	 * @return FreshRSS_StatsDAOSQLite|FreshRSS_StatsDAOPGSQL|FreshRSS_StatsDAO
```
Comment (Alkarex) [inline app/Models/Factory.php]: `FreshRSS_TagDAO` should be enough

File: app/Models/EntryDAO.php (hunk 1 above)
Comment (ColonelMoutarde) [reply app/Models/EntryDAO.php]: your right ! I have forgotten

Comment (ColonelMoutarde) [reply app/Models/EntryDAO.php]: i remember... I focused on Factory.php this file does not need to be fully modified for phpstan to pass. My target is factory.php

Comment (Alkarex) [reply app/Models/EntryDAO.php]: Yes fine, but then please avoid reformatting

Comment (ColonelMoutarde) [reply app/Models/EntryDAO.php]: done

File: app/Models/Factory.php (hunk 2 above)
Comment (ColonelMoutarde) [reply app/Models/Factory.php]: good remark, done

File: app/Controllers/entryController.php [hunk 3]
```diff
@@ -78,10 +78,10 @@ public function readAction() {
 				$get = substr($get, 2);
```
Comment (Alkarex) [inline app/Controllers/entryController.php]:
```suggestion
				$get = (int)substr($get, 2);
```

And remove the casts further down

File: app/Models/Factory.php [hunk 4]
```diff
@@ -15,7 +24,12 @@ public static function createCategoryDao($username = null) {
 		}
 	}
 
-	public static function createFeedDao($username = null) {
+	/**
+	 * @return FreshRSS_FeedDAOSQLite|FreshRSS_FeedDAO
```
Comment (Alkarex) [inline app/Models/Factory.php]:
```suggestion
	 * @return FreshRSS_FeedDAO
```

Returning the top parent class should be sufficient. Same for other lines below

**Pull request merged**